src/common/helpers/sharepoint/sharepoint.js

```diff
@@ -13,69 +13,114 @@ import {
 } from '../../connectors/notify/notify.js'
 import { escapeMarkdown } from '../escape-text.js'
 import { createSharepointItem } from './sharepoint-item.js'
+import { sendMessageToSQS } from '../../connectors/queue/sqs.js'
+import { createLogger } from '../logging/logger.js'
 
 /**
  * @import {FileAnswer, ApplicationData} from '../../../common/helpers/data-extract/data-extract.js'
  * @import {HandlerError} from '../../../common/helpers/types.js'
+ * @typedef {{ application: ApplicationData, reference: string }} QueuedApplication
  */
 
+const logger = createLogger()
+
 /**
  * @param {object} request
  * @param {string} reference
  * @returns {Promise<void|HandlerError>}
  */
 export const sharePointApplicationHandler = async (request, reference) => {
-  // Upload the application to SharePoint
-  const applicationHtml = await generateHtmlBuffer(request.payload, reference)
   try {
-    await uploadFile(
-      reference,
-      `${reference}_Submitted_Application.html`,
-      applicationHtml
-    )
+    await sendMessageToSQS(request.payload, reference)
   } catch (error) {
-    request.logger.warn(`Failed to upload file to SharePoint: ${error.message}`)
     return {
       error: {
-        errorCode: 'FILE_UPLOAD_FAILED__APPLICATION',
+        errorCode: 'MESSAGE_ENQUEUEING_FAILED',
         statusCode: statusCodes.serverError
       }
     }
   }
+  try {
+    await sendApplicantConfirmationEmail(request.payload, reference)
+  } catch (error) {
+    logger.error(`Failed to send email to applicant: ${error.message}`)
+  }
+}
 
-  // Upload the biosecurity map if it exists
+/**
+ * @param {QueuedApplication} queuedApplicationData
+ * @returns {Promise<void|HandlerError>}
+ */
+export const processApplication = async (queuedApplicationData) => {
+  const { reference, application } = queuedApplicationData
+
+  try {
+    await uploadSubmittedApplication(application, reference)
+  } catch (error) {
+    logger.warn(
+      `Failed to upload submitted application to SharePoint: ${error.message}`
+    )
+    throw new Error(error)
+  }
+
+  try {
+    await uploadBiosecurityMap(application, reference)
+  } catch (error) {
+    logger.warn(
+      `Failed to upload biosecurity map to SharePoint: ${error.message}`
+    )
+    throw new Error(error)
+  }
+
+  let item
+  try {
+    item = await createSharepointItem(application, reference)
+  } catch (error) {
+    logger.warn(`Failed to create SharePoint item: ${error.message}`)
+    throw new Error(error)
+  }
+  try {
+    await sendCaseworkerNotificationEmail(application, reference, item)
+  } catch (error) {
+    logger.warn(`Failed to send email to case worker: ${error.message}`)
+    throw new Error(error)
+  }
+  return undefined
+}
+
+/**
+ * @param {ApplicationData} application
+ * @param {string} reference
+ * @returns {Promise<void>}
+ */
+const uploadSubmittedApplication = async (application, reference) => {
+  const applicationHtml = await generateHtmlBuffer(application, reference)
```

Comment:
This is strictly speaking correct, but `generateHtmlBuffer` doesn't need to be an async function necessarily. 

Not a great deal of harm in this, though.

Reply:
done